### PR TITLE
[autoscaler][kubernetes][docs] Updated Kubernetes Documentation

### LINKED ISSUE #13655: Ray dashboard doesn't work with K8s operator
### What is the problem?

*Ray version and other system information (Python version, TensorFlow version, OS):*

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

### LINKED ISSUE #13658: [autoscaler][kubernetes][docs] Simplify Kubernetes documentation
There are three pages of docs. It's pretty confusing.

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Why are these changes needed?

This PR updates Ray's Kubernetes documentation. The documentation is by no means comprehensive yet and could use to be iterated on.

However it does streamline the K8s documentation a bit and provides critical information that wasn't clearly explained before (esp. how to use Ray Client). 

- The main "Deploying on Kubernetes" documentation now explains how to run an autoscaling Ray cluster with the Cluster Launcher and Operator. There's an overview and a comparison of the two launching methods.
- There's an updated explanation of how to connect to the Dashboard and Client to match this recent commit: https://github.com/ray-project/ray/pull/13920
- Manual Deployment launching instructions and notes on GPU usage are moved into subpages under the main "Deploying on Kubernetes" pages. There are also links to these two documentation pages from the top of the main page.

Note: We've decided to merge these docs directly into master rather than into this documentation branch: https://github.com/ray-project/ray/pull/13839
All substantial revisions are contained within the "Deploying on Kubernetes" under "Ray with Cluster Managers", so this PR is pretty self-contained.

## Related issue number

Closes https://github.com/ray-project/ray/issues/13655
Closes https://github.com/ray-project/ray/issues/13658

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/kubernetes.rst [hunk 1]
```diff
@@ -1,271 +1,464 @@
...
+
+
+Overview
+========
+The Ray Autoscaler uses the `NodeProvider interface`_ to manage Ray nodes.
+To autoscale in different contexts (e.g. on AWS, GCP, Azure, Kubernetes) the Autoscaler uses subclasses of this interface.
+The AWS, GCP, and Azure node providers provision one cloud instance per Ray node.
+
+The Kubernetes node provider on the other hand, spins up one Kubernetes pod per Ray node.
+Thus, when using the Ray Autoscaler with Kubernetes, each Ray node runs in its `own Kubernetes pod <https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/_private/kubernetes/node_provider.py>`__.
```
Comment (richardliaw) [inline doc/source/cluster/kubernetes.rst:19]: This is quite hard to parse -- 
1. Users don't need to know the existence of a Node Provider
2. Users will most care about "what is the K8s - Ray story and how do I get started"

```suggestion
```

File: doc/source/cluster/kubernetes-manual.rst [hunk 2]
```diff
@@ -0,0 +1,160 @@
+.. _ray-k8s-deploy:
+
+Deploying a Static Cluster
+===================================
```
Comment (richardliaw) [inline doc/source/cluster/kubernetes-manual.rst:4]:
```suggestion
Deploying a Static Cluster
==========================
```

File: doc/source/cluster/kubernetes-manual.rst [hunk 3]
```diff
@@ -0,0 +1,160 @@
+.. _ray-k8s-deploy:
```
Comment (richardliaw) [inline doc/source/cluster/kubernetes-manual.rst:1]: should we change this tag?

File: doc/source/cluster/kubernetes-gpu.rst [hunk 4]
```diff
@@ -0,0 +1,89 @@
+.. _k8s-gpus:
+
+GPU Usage
```
Comment (richardliaw) [inline doc/source/cluster/kubernetes-gpu.rst:3]:
```suggestion
GPU Usage with Kubernetes
```

File: doc/source/cluster/kubernetes.rst [hunk 5]
```diff
@@ -1,271 +1,464 @@
...
+***********************
 Deploying on Kubernetes
-=======================
+***********************
+
+.. toctree::
+    :hidden:
+
+    /cluster/kubernetes-manual.rst
+    /cluster/kubernetes-gpu.rst
```
Comment (richardliaw) [inline doc/source/cluster/kubernetes.rst:9]: can you just orphan these two files? They mess with the sidebar (on the left side)

File: doc/source/cluster/kubernetes.rst [hunk 6]
```diff
@@ -1,271 +1,464 @@
...
+Ray supports two ways of launching an autoscaling Ray cluster on Kubernetes.
+
+- Using the :ref:`Ray Cluster Launcher <k8s-cluster-launcher>`
+- Using the :ref:`Ray Kubernetes Operator <k8s-operator>`
+
+
+.. _`NodeProvider interface`: https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/node_provider.py
+
+
+Comparision of the Cluster Launcher and Operator
```
Comment (richardliaw) [inline doc/source/cluster/kubernetes.rst:30]: Before talking about a comparison of the two, you need to first start with answering WHAT are the two?

File: doc/source/cluster/kubernetes.rst [hunk 7]
```diff
@@ -1,271 +1,464 @@
...
+   * On the other hand, the Operator's ``headRayStartCommands`` should include a ``--no-monitor`` flag to prevent the autoscaling/monitoring process from running on the head node.
+
+
+Quick Guide
+-----------
+
+The next two sections detail how to launch with the :ref:`Ray Cluster Launcher <k8s-cluster-launcher>`
+and with the :ref:`Ray Kubernetes Operator <k8s-operator>`, respectively.
+
+The section after that describes how to :ref:`interact <ray-k8s-interact>` with your Ray cluster
```
Comment (richardliaw) [inline doc/source/cluster/kubernetes.rst:74]: Can you move this quick guide to the top (underneath "Ray supports ... bullet points")? This is basically a table of contents.

A list format would also be nice here (since it's currently just oddly spaced)

**Review by richardliaw — CHANGES_REQUESTED** [review]
Left a couple comments, but it wasn't a thorough read (mainly focused on front matter).

Happy to do another pass after addressed!

**Comment by DmitriGekhtman** [comment]
Thanks, @richardliaw . Could you have another look?

File: doc/source/cluster/kubernetes.rst [hunk 8]
```diff
@@ -282,114 +459,143 @@ the following:
...
+  root@ray-head-6f566446c-5rdmb:/# python /home/ray/anaconda3/lib/python3.7/site-packages/ray/autoscaler/kubernetes/example_scripts/run_on_head.py
+  # You should see repeated output for 10 iterations and then 'Success!'
+
+
+The program in this example uses ``ray.init(address="auto")`` to connect to the Ray cluster.
+
+Accessing the Dashboard
+-----------------------
+
+The Ray Dashboard can accessed locally using ``kubectl port-forward``.
```
Comment (richardliaw) [inline doc/source/cluster/kubernetes.rst:527]: link to ray dashboard page

File: doc/source/cluster/kubernetes.rst [hunk 9]
```diff
@@ -1,271 +1,448 @@
...
+- Using the :ref:`Ray Kubernetes Operator <k8s-operator>`
 
-.. _`Ray repository`: https://github.com/ray-project/ray/tree/master/doc/kubernetes
+The Cluster Launcher and Ray Kubernetes Operator provide similar functionality; each serves as an `interface to the Ray autoscaler`.
+Below is a brief overview of the two tools.
 
-Creating a Ray Namespace
+The Ray Cluster Launcher
 ------------------------
+The Ray Cluster Launcher is geared towards experimentation and development.
```
Comment (richardliaw) [inline doc/source/cluster/kubernetes.rst:44]: Link to Ray cluster launcher page

Comment (richardliaw) [reply doc/source/cluster/kubernetes.rst:44]:
```suggestion
The Ray Cluster Launcher is geared towards experimentation and development and can be used to launch Ray clusters on Kubernetes (among other backends).
```

File: doc/source/cluster/kubernetes.rst [hunk 10]
```diff
@@ -1,271 +1,448 @@
...
+   The Ray Kubernetes Operator requires Kubernetes version at least ``v1.17.0``. Check Kubernetes version info with the command
+   :bash:`kubectl version`.
+
+.. note::
+   The example commands in this document launch six Kubernetes pods, using a total of 6 CPU and 3.5Gi memory.
+   If you are experimenting using a test Kubernetes environment such as `minikube`_, make sure to provision sufficient resources, e.g.
+   :bash:`minikube start --cpus=6 --memory=\"4G\"`.
+   Alternatively, reduce resource usage by editing the ``yaml`` files referenced in this document; for example, reduce ``minWorkers``
+   in ``example_cluster.yaml`` and ``example_cluster2.yaml``.
+
```
Comment (richardliaw) [inline doc/source/cluster/kubernetes.rst:141]: this is a lot of warning/notes. Can you fuse them or split them somehow?

**Review by richardliaw — APPROVED** [review]
A huge fan of these new docs! Thanks a bunch for doing this -- left some last comments. Great work, @DmitriGekhtman !

**Comment by DmitriGekhtman** [comment]
@richardliaw 
Thanks again for the helpful comments -- could you look over it once more and merge if it looks ok?

**Comment by richardliaw** [comment]
@DmitriGekhtman lint is failing

**Comment by richardliaw** [comment]
ping when lint passes (currently not passing)

**Comment by DmitriGekhtman** [comment]
> ping when lint passes (currently not passing)

blrrrg, i noticed

**Comment by DmitriGekhtman** [comment]
Having trouble fixing the lint error:

Anyone have a sense of what's causing links to references `k8s-gpus` and `ray-k8s-static` to fail to compile?

**Comment by DmitriGekhtman** [comment]
> Having trouble fixing the lint error:
> 
> Anyone have a sense of what's causing links to references `k8s-gpus` and `ray-k8s-static` to fail to compile?

Never mind, I see the problem.

**Comment by DmitriGekhtman** [comment]
@richardliaw lint passed
